linker/shared/src/main/scala/org/scalajs/linker/backend/javascript/Printers.scala

```diff
@@ -842,4 +842,10 @@ object Printers {
     }
   }
 
+  /** A printer that shows `Tree`s for debugging, not for pretty-printing. */
+  class JSTreeShowPrinter(_out: ByteArrayWriter, initIndent: Int = 0)
```

Comment:
I think this makes sense. A couple of higher level points:

- Should we expose this as a method `showTree(Tree): String` only?
- I think at this point `JSTreePrinter.printTree` should probably be made protected?